common/app/routes/challenges/components/Classic-Modal.jsx

```diff
@@ -0,0 +1,86 @@
+import React, { PropTypes } from 'react';
+import { Button, Modal } from 'react-bootstrap';
+import PureComponent from 'react-pure-render/component';
+import FontAwesome from 'react-fontawesome';
+
+const propTypes = {
+  close: PropTypes.func,
+  keyMap: PropTypes.object,
+  mapKeys: PropTypes.func,
+  open: PropTypes.bool.isRequired,
+  submitChallenge: PropTypes.func.isRequired,
+  successMessage: PropTypes.string.isRequired
+};
+
+export default class ClassicModal extends PureComponent {
+  constructor(...props) {
+    super(...props);
+    this.keyBindings = this.keyBindings.bind(this);
+  }
+  keyBindings(event) {
+    const map = Object.assign({}, this.props.keyMap);
+    const { open, submitChallenge, mapKeys } = this.props;
+    if (event.metaKey) {
+      map['meta'] = event.type === 'keydown';
+    }
+    map[event.keyCode] = event.type === 'keydown';
+    // (Ctrl || Cmd) + Enter
+    if ((map[17] || map['meta']) && map[13] && open) {
+      submitChallenge();
+    }
+    mapKeys(map);
+  }
+
+
+  render() {
+    let {
+      close,
+      open,
+      submitChallenge,
+      successMessage
+    } = this.props;
+    return (
+      <Modal
+        animation={ false }
+        dialogClassName='success'
+        keyboard={ true }
+        onHide={ close }
+        onKeyDown={ this.keyBindings }
+        onKeyUp={ this.keyBindings }
```

Comment:
This is becoming incredibly complicated for a simple event listener.

> This needs to be called on keyup also, to set the keyMap prop to false.

Removing the keys from state solve the issue of needing to reset state.

> This is so that a user cannot press Ctrl || Cmd and then Enter individually to submit the challenge. The Ctrl and Enter both need to be keydown to submit the challenge.

(ctrl || cmd) are modifying the enter keypress. We don't listen for their individual key events. We only care that the enter key is pressed and that the ctrl or cmd key are also pressed, modifying the meaning of the enter key. If they are not pressed at the same time then their value's will be false. This works everywhere else in the codebase.

This is what the event handler should look like:

```js
  handleKeyDown(e) {
   if (
    e.keyCode === enterKey &&
    (e.meta || e.ctrlKey)
   ) {
      e.preventDefault();
      this.props.submitChallenge();
    }
  }
```

When the modal is in focus, (ctrl || cmd) + enter will trigger the submitChallenge action and prevent the key event from bubbling up. Otherwise it does nothing.
No need to add to state or track previously clicked buttons. Is there something that prevents this from working?